core/workflow/loadDPL.go

```diff
@@ -25,13 +25,21 @@
 package workflow
 
 import (
+	"gopkg.in/yaml.v3"
+	"strings"
+
 	"github.com/AliceO2Group/Control/common/gera"
 	"github.com/AliceO2Group/Control/core/task"
 )
 
-func LoadDPL(tasks []*task.Class) (workflow Role, err error) {
+func LoadDPL(tasks []*task.Class, filename string) (workflow Role, err error) {
+	// FIXME: base roleBase of root defaults to all empty values
 	root := new(aggregatorRole)
 
+	if strings.Contains(filename, ".json"){
+		root.roleBase.Name = filename[:len(filename)-5]
+	}
+
```

Comment:
I think generating the filename and/or root role name should happen in the caller, outside the `workflow` package. So `func LoadDPL(tasks []*task.Class, rootRoleName string) (workflow Role, err error)`.